### PR TITLE
spacebar down for hand tool + cleanSceneView

### LINKED ISSUE #121: [feature] pressing space bar (fill mode) also activates the hand tool for panning
Hey, I just tried this for the first time and I love it already!

I have a UI/UX suggestion. It'd be nice if, while pressing the space bar and in black-on-white fill mode, the hand tool were also activated such that one can pan around by dragging the cursor.
I think also other font editors do this (I forgot if it's Glyphs.app or Robofont or both, don't have a mac right now to confirm).

Anyway, thanks for considering.

### PR DESCRIPTION
**Pull request by ollimeier**
Fixes #121

### CODE REVIEW

File: src/fontra/views/editor/editor.js [hunk 1]
```diff
@@ -611,6 +611,7 @@ export class EditorController {
   }
 
   initTools() {
+    this.toolIdentifierSelected = undefined;
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:614]: Please call this `selectedToolIdentifier`.

File: src/fontra/views/editor/editor.js [hunk 2]
```diff
@@ -873,7 +874,10 @@ export class EditorController {
     };
   }
 
-  setSelectedTool(toolIdentifier) {
+  setSelectedTool(toolIdentifier, setToolIdentifier = true) {
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:877]: This new argument is not needed if we rearrange the logic a little bit.

I think the space-down and space-up code should be themselves be responsible for remembering which tool was selected. I will comment more below.

In the meantime, `this.selectedToolIdentifier` should be set at the end of this method, right after `this.sceneController.setSelectedTool(this.tools[selectedToolIdentifier]);`, and it should use the `selectedToolIdentifier` variable.

File: src/fontra/views/editor/editor.js [hunk 3]
```diff
@@ -1938,6 +1942,7 @@ export class EditorController {
     for (const overlay of document.querySelectorAll(".cleanable-overlay")) {
       overlay.classList.add("overlay-layer-hidden");
     }
+    this.setSelectedTool("hand-tool", false);
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:1945]: Then, here we should remember the currently selected tool, so we can restore it in the space-up code.

I suggest this:
```js
      this.savedSelectedToolIdentifier = this.selectedToolIdentifier;
      this.setSelectedTool("hand-tool");
```

File: src/fontra/views/editor/editor.js [hunk 4]
```diff
@@ -1946,6 +1951,7 @@ export class EditorController {
     for (const overlay of document.querySelectorAll(".cleanable-overlay")) {
       overlay.classList.remove("overlay-layer-hidden");
     }
+    this.setSelectedTool(this.toolIdentifierSelected);
```
Comment (justvanrossum) [inline src/fontra/views/editor/editor.js:1954]: Then here, we do
```js
    this.setSelectedTool(this. savedSelectedToolIdentifier);
    delete this.savedSelectedToolIdentifier;
```

This in total is a bit more code, but the intention is much much clearer.

Comment (ollimeier) [reply src/fontra/views/editor/editor.js]: Done.

File: src/fontra/views/editor/editor.js (hunk 3 above)
Comment (ollimeier) [reply src/fontra/views/editor/editor.js]: Done.

File: src/fontra/views/editor/editor.js [hunk 5]
```diff
@@ -611,6 +611,7 @@ export class EditorController {
   }
 
   initTools() {
+    this.selectedToolIdentifier = undefined;
```
[the remark this answers is not in the document]
Comment (ollimeier) [reply src/fontra/views/editor/editor.js]: Thanks. Done.

**Pull request merged**